docs/proposals/hpa/federated-hpa.md

```diff
@@ -0,0 +1,333 @@
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? 
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+### Architecture
+![Karmada FederatedHPA Architecture](statics/karmadafederatedhpa-arch.drawio.png)
+
+There are no new CRDs or resources introduced in this design. All the core functions are implemented in the `FederatedHPAController`.
+1. The Kubernetes HPA components are still used in the member cluster and can work standalone.
+1. The FederatedHPAController is responsible for the purposes
+   1. Watch the HPA resource and `PropagationPolicy/ResourceBinding` corresponding to the `Workload`, to learn which clusters the HPA resource should propagated to and what weight the workloads should be spread between clusters.
+   1. Create `PropagationPolicy` and `OverridePolicy` corresponding to HPA resource to spread the HPA to clusters. Distribute `min/max` fields of the HPA resources between member clusters based on the weight learned.
+   1. Redistribute 'some' fields of the HPA resources after disaster of member clusters detected or `PropagationPolicy/ResourceBinding` corresponding to `Workload` is changed.
+1. The `Propagation`s corresponding to the relevant resources(HPA, the workloads the HPA targeted and the FederatedQuota) should coincide or should be the same. That means, if the `Workload` should be propagated to `clustera` and `clusterb`, the corresponding HPA and federated Quota resources should be propagated to `clustera` and `clusterb` too.
+
+### How FederatedHPAController learn the propagation information corresponding to the Workload
+When a new HPA resource created or changed, the `FederatedHPAController` should know the propagation and weight information of the corresponding `Workload`. How does the `FederatedHPAController` know it? The `FederatedHPAController` will easily find the corresponding `Workload` based on the field `ScaleTargetRef` and then will find the `PropagationPolicy` resource based on the matching of `ResourceSelectors`. 
+
+For the weight information, because the karmada scheduler already plays the role to schedule the replicas, the `FederatedHPAController` can simplely reuse the scheduling result to learn the weight. The `HPAController` in the member cluster scale the `Workload` in the member cluster directly, it will conflict between the karmada scheduler. Maybe we can introduce `autoscaling/manualscaling` mode represented by the annotation `policy.karmada.io/scalingmode=[autoscaling|manualscaling]` in the `ResourceBinding` consuming by karmada scheduler and binding controller. In the `autoscaling` mode, the karmada scheduler should not schedule the replicas directly but calculate the weight information instead and the binding controller should ignore the replicas override.
+
```

Comment:
the binding controller should ignore the replicas override
it should be updated, then binding controller will update work resources. Only in this way, replicas of the work is the same as replicas of the member cluster. Otherwise, the execution controller will update the old replicas to the member cluster. the workload will keep scale up，then scale down in memeber cluster. it will repeat these steps.

Reply:
@riverzhang Thanks for your comment. You are right, it wouldn't work if we simply remove the replicas override in the binding controller. After discussion with @snowplayfire , we find that we can use [resource interpreter webhook](https://github.com/karmada-io/karmada/tree/master/docs/proposals/resource-interpreter-webhook) to retain the replicas in the member cluster. I will update the proposal later.